spec/an_metadata_update_helper_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'tmpdir'
+require_relative './spec_helper'
+
+describe Fastlane::Helper::StandardMetadataBlock do
+  it 'strips any trailing newline when generating the block for a single-line input' do
+    Dir.mktmpdir do |dir|
+      input = "Single line message with new line\n"
+
+      # Generate the input file to convert to .pot block
+      input_path = File.join(dir, 'input')
+      File.write(input_path, input)
+      # Ensure the input has only one line
+      expect(File.read(input_path).lines.count).to eq 1
+
+      # Write the .pot block in a StringIO to bypass the filesystem and have
+      # faster tests
+      output_io = StringIO.new
+      described_class.new('any-key', input_path).generate_block(output_io)
+
+      # Ensure the output matches the expectation: the trailing new line has
+      # been stripped.
+      #
+      # Note that the final new line is intentional. It's part of the
+      # formatting at the time of writing.
+      expect(output_io.string).to eq <<~EXP
+        msgctxt "any-key"
+        msgid "Single line message with new line"
+        msgstr ""
+
+      EXP
+    end
+  end
+
+  it 'does not strip a trailing new line when generating the block for a multi-line input' do
+    Dir.mktmpdir do |dir|
+      input = "Multi-line\nmessage\nwith\ntrailing new line\n"
+
+      # Generate the input file to convert to .pot block
+      input_path = File.join(dir, 'input')
+      File.write(input_path, input)
+
+      # Write the .pot block in a StringIO to bypass the filesystem and have
+      # faster tests
+      output_io = StringIO.new
+      described_class.new('any-key', input_path).generate_block(output_io)
+
+      # Ensure the output matches the expectation: the trailing new line has
+      # been stripped.
+      #
+      # Note that the final new line is intentional. It's part of the
+      # formatting at the time of writing.
+      expect(output_io.string).to eq %q(msgctxt "any-key"
+msgid ""
+"Multi-line\n"
+"message\n"
+"with\n"
+"trailing new line\n"
+msgstr ""
+
+)
```

Comment:
I would have liked to use the heredoc syntax like in the other test to avoid the difference in indentation, but that syntax escapes characters, and the comparison against the expected `\n` fails.

Reply:
FYI, since Ruby 2.3 you can use `<<~'EXPECTED'` – with single-quotes around the heredoc identifier – to prevent interpolation of the content. (and the `~` is to allow indentation of the content to be ignored.

```suggestion
      expect(output_io.string).to eq <<~'EXPECTED'
        msgctxt "any-key"
        msgid ""
        "Multi-line\n"
        "message\n"
        "with\n"
        "trailing new line\n"
        msgstr ""
EXPECTED
```

See https://rubyreferences.github.io/rubyref/language/literals.html#here-documents-heredocs

This is just a nitpick so not a blocker for this to be merged, but figured you'd be curious to know 😉

Reply:
Oh YES! I got close to that syntax in my research for a solution yesterday, but not quite there.